config/manager/manager.yaml

```diff
@@ -22,7 +22,6 @@ spec:
       containers:
         - args:
             - --leader-elect
-            - "--metrics-bind-addr=127.0.0.1:8080"
             - "--feature-gates=MachinePool=${EXP_MACHINE_POOL:=false},AKS=${EXP_AKS:=false}"
```

Comment:
just fyi: In CAPI we kept that one (but with `localhost:8080`) so we can replace it in our e2e test (https://github.com/kubernetes-sigs/cluster-api/blob/89df9472d1678ed8ff59f4e124d48ad51ffd85c6/test/e2e/config/docker.yaml#L53-L54) and the e2e test framework is able to retrieve the metrics via proxy (https://github.com/kubernetes-sigs/cluster-api/blob/c55b5013fcb84841cc3705dfb250cdfd8b140447/test/framework/deployment_helpers.go#L200-L206)

Not sure if that's also relevant for CAPZ

Reply:
added back with localhost

Reply:
If you want to collect metrics during e2e tests you also have to add the replacements to `azure-dev.yaml` (for all providers).

Otherwise you'll get: https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/1730/pull-cluster-api-provider-azure-e2e/1442611507668979712/artifacts/clusters/bootstrap/controllers/capi-controller-manager/capi-controller-manager-7dc56b9799-mk5tf/metrics-error.txt